applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp

```diff
@@ -650,6 +660,45 @@ class EigensolverStrategy
         KRATOS_CATCH("")
     }
 
+    /// Normalize the eigenmodes with respect to the mass matrix
+    void MassNormalizeEigenvectors(DenseMatrixType& rEigenvectors)
+    {
+        const SparseMatrixType& rMassMatrix = this->GetMassMatrix();
+        const std::size_t num_modes = rEigenvectors.size1();
+        const std::size_t num_dofs  = rEigenvectors.size2();
+
+        DenseVectorType phi(num_dofs);
+        DenseVectorType tmp(num_dofs);
+
+        for (std::size_t j = 0; j < num_modes; ++j) {
+             // copy row j -> phi
+            IndexPartition<std::size_t>(num_dofs).for_each([&](std::size_t i){
+                phi[i] = rEigenvectors(j, i);
+            });
+
+            // tmp = M * phi_j
+            TSparseSpace::Mult(rMassMatrix, phi, tmp);
+
+            // Compute modal_mass = phi^T * tmp
+            const double modal_mass = IndexPartition<std::size_t>(num_dofs).for_each<SumReduction<double>>([&](std::size_t i)->double {
```

Comment:
```suggestion
            const double modal_mass = IndexPartition<std::size_t>(num_dofs).for_each<SumReduction<double>>([&phi, &tmp](std::size_t i)->double {
```

I would only capture what is required only.

Reply:
Btw, I think the eigen mode solver supports MPI ryt? If so, in anycase I would do the following

```c++
    const auto global_modal_mass = rModelPart.GetCommunicator().GetDataCommunicator().SumAll(modal_mass);
```

Reply:
Done

Reply:
It doesn't support MPI

Neither our code, nor the eigensolvers we integrate

Trilinos has Anasazi I think, but we never tried

Reply:
So, what do we do @sunethwarna @matekelemen?

Reply:
Good as is, the SumAll doesn't hurt